crates/turborepo-lsp/src/lib.rs

```diff
@@ -0,0 +1,853 @@
+            .expect("only fails if poisoned")
+            .clone();
+
+        let repo_root = match repo_root {
+            Some(repo_root) => repo_root,
+            None => {
+                self.client
+                    .log_message(MessageType::INFO, "received request before initialization")
+                    .await;
+                return;
+            }
+        };
+
+        let packages = self.package_discovery().await;
+
+        let tasks = packages.map(|p| {
+            p.workspaces
+                .into_iter()
+                .filter_map(|wd| {
+                    let package_json = PackageJson::load(&wd.package_json).ok()?; // if we can't load a package.json, then we can't infer its tasks
+                    let package_json_name = if repo_root.contains(&wd.package_json) {
+                        Some("//".to_string())
+                    } else {
+                        package_json.name
+                    };
+                    Some(
+                        package_json
+                            .scripts
+                            .into_keys()
+                            .map(move |k| (k, package_json_name.clone())),
+                    )
+                })
+                .flatten()
+                .into_group_map()
+        });
+
+        // we still want to emit diagnostics if we can't infer tasks
+        let tasks_and_packages = tasks.as_ref().map(|tasks| {
+            (
+                tasks,
+                tasks
+                    .values()
+                    .flatten()
+                    .flatten()
+                    .map(|s| s.as_str())
+                    .unique()
+                    .collect::<HashSet<_>>(),
+            )
+        });
+
+        let mut diagnostics = vec![];
+
+        // ParseResult cannot be sent across threads, so we must ensure it is dropped
+        // before we send the diagnostics. easiest way is just to scope it
+        {
+            let parse =
+                jsonc_parser::parse_to_ast(&contents, &Default::default(), &Default::default());
+
+            let parse = match parse {
+                Ok(parse) => parse,
+                Err(_) => return, // if it is not a valid json, then there are no diagnostics
+            };
+
+            let object = parse.value.as_ref().and_then(|v| v.as_object());
+
+            let mut globs = vec![];
+
+            globs.extend(
+                object
+                    .and_then(|o| o.get_array("globalDependencies"))
+                    .map(|d| &d.elements)
+                    .into_iter()
+                    .flatten(),
+            );
+
+            let pipeline = object
+                .and_then(|o| o.get_object("pipeline"))
+                .map(|p| p.properties.iter());
+
+            for property in pipeline.into_iter().flatten() {
+                let (package, task) = property
+                    .name
+                    .as_str()
+                    .rsplit_once('#')
```

Comment:
We don't really support tasks with multiple `#`'s, but just so we're consistent in where we chop them. [Rust](https://github.com/vercel/turbo/blob/main/crates/turborepo-lib/src/run/task_id.rs#L131) [Go](https://github.com/vercel/turbo/blob/main/cli/internal/util/task_id.go#L30)
```suggestion
                    .split_once('#')
```